synths/default-synths.scd

```diff
@@ -253,4 +253,130 @@ SynthDef(\supersiren, {|out, sustain, pan, n |
 	OffsetOut.ar(out, DirtPan.ar(sound, ~dirt.numChannels, pan, env))
 }).add;
 
+// The next four synths respond to the following parameters in addition to gain, pan, n, and all the "effect" parameters
+// (including attack, hold, and release).  Default values in parentheses.
+//
+// sustain - scales overall duration
+// decay(0) - amount of decay after initial attack
+// accelerate(0) - pitch glide
+// semitone(12) - how far off in pitch the secondary oscillator is (need not be integer)
+// pitch1(1) - filter frequency scaling multiplier, the frequency itself follows the pitch set by "n"
+// speed(1)- LFO rate
+// lfo(1) - how much the LFO affects the filter frequency
+// resonance(0.2) - filter resonance
+// voice(0.5) - depends on the individual synth
+
+// a moog-inspired square-wave synth; variable-width pulses with filter frequency modulated by an LFO
+// "voice" controls the pulse width (exactly zero or one will make no sound)
+SynthDef(\supersquare, {|out, speed=1, decay=0, sustain, pan, accelerate, n,
+	   voice=0.5, semitone=12, resonance=0.2, lfo=1, pitch1=1|
+	var env = EnvGen.ar(Env.pairs([[0,0],[0.05,1],[0.2,1-decay],[0.95,1-decay],[1,0]], -3), timeScale:sustain, doneAction:2);
+	var basefreq = n.midicps * Line.kr(1, 1+accelerate, sustain);
```

Comment:
instead of using `n`, you could use the new `freq` parameter to calculate the basefreq:

`freq * Line.kr(1, 1+accelerate, sustain);`

Then if someone changes the pitch calculation in superdirt, it will be used by this synth.